src/writeLAS.cpp

```diff
@@ -172,6 +172,24 @@ void C_writer(CharacterVector file, List LASheader, List data)
             ExtraByte.start = header.get_attribute_start(ExtraByte.id);
         }
       }
+      else if (names[i] == "TextArea")
+      {
+        if (vlr.containsElementNamed("Text Area Description"))
+        {
+          CharacterVector ASCII = vlr["Text Area Description"];
+          std::string sascii    = as<std::string>(ASCII);
+          CharacterVector desc  = vlr["description"];
+          std::string sdesc  = as<std::string>(desc);
+          I32 sascii_size = sascii.size();
+          if ((sizeof(CHAR)*sascii_size) > U16_MAX)
```

Comment:
Be careful if you truncate the string it must be null-terminated

Reply:
I have added a null terminator to text of any length as it is not done by LASlib.
Therefore, `length_after_header == nchar(vlr$text_area_description)+1` always.